src/main/java/com/exadel/discountwebapp/discount/mapper/DiscountMapper.java

```
@@ -65,9 +79,14 @@ public void updateEntity(DiscountRequestVO request, Discount discount) {
     }
 
     private void provideDiscountDependencies(DiscountRequestVO request, Discount discount) {
-        Vendor vendor = vendorRepository.findById(request.getVendorId()).orElse(null);
-        Category category = categoryRepository.findById(request.getCategoryId()).orElse(null);
+        var vendor = vendorRepository.findById(request.getVendorId()).orElse(null);
```

throw EntityNotFoundException instead of returning null

---

Thanks, fixed